### DISCUSSION TITLE
两个组件不在一块，怎么在别的组件中拿到FormDialog的关闭事件

### DISCUSSION THREAD

**Discussion by beita-wang** [alibaba/formily#3788] [Q&A 2.x]
FormDialog 里套了个Table,双击Table的某一行，然后关闭FormDialog。 用的是schema模式，怎么在列表中拿到FormDialog的关闭事件。

**Comment by luqiudi** [comment]
你在当前组件里面用useField() 获取当前字段,然后可以获取到他的父字段,然后你怎么操作都行了.

import { useField } from "@formily/vue"

const field = useField()

//父字段的内容
field.value.parent
field.value.parent.component

**Reply by beita-wang** [reply]
我指的是怎么在别的组件中拿到FormDialog的关闭事件，而不是拿到Field。FormDialog是一个api。
已解决：使用Context，放入formDialog实例。

**Comment by beita-wang** [comment]
使用Context，放入formDialog实例。
